vslam/feature/superpoint_feature.py

```diff
@@ -0,0 +1,80 @@
+from typing import List, Tuple, Optional
+import numpy as np
+import cv2
+from .feature import FeatureExtractor
+from .SuperPointPretrainedNetwork.demo_superpoint import SuperPointFrontend, PointTracker, VideoStreamer
+
+class SuperPointFeatureExtractor(FeatureExtractor):
+    """
+    Feature Extractor for SuperPoint
+    """
+    def __init__(self, **kwargs) -> None:
+        """Contructor
+            TODO: SuperPoint natively has trackers for points matched across all frames
+        """
+        super().__init__()
+        self.img_h = kwargs['img_h']
+        self.img_w = kwargs['img_w']
+        
+        self.fe = SuperPointFrontend(
+                        weights_path=kwargs['weights_path'],
+                        nms_dist=kwargs['nms_dist'],
+                        conf_thresh=kwargs['conf_thresh'],
+                        nn_thresh=kwargs['nn_thresh'],
+                        cuda=kwargs['cuda'])
+
+    def detect(self, img: np.ndarray, mask: np.ndarray = None) -> List:
+        """Wrapper class for SIFT detector
+            Args:
+                Image
+                Mask
+
+            Returns:
+                List of keypoints
+        """
+        pass
+
+    def compute(self, img: np.ndarray, kp: List) -> Tuple[List, np.ndarray]:
+        """Wrapper class for SIFT compute
+            Args:
+                Image
+
+            Returns:
+                Tuple of keypoint and corresponding feature descriptor
+        """
+        pass
+
+    def _pts_to_keypoints(self, pts : np.ndarray, orig_shape : Tuple, resized_shape : Tuple) -> List:
```

Comment:
maybe makes sense to add these types of functions into a separate module as vslam utils? this function is also written in Tracker as far as I remember.